optimum/intel/openvino/quantization.py

```
@@ -584,4 +588,98 @@ def _weight_only_quantization(
+    if input_tensor.get_element_type().get_type_name() in allowed_types_list:
+        const_node = _get_operation_const_op(node, const_port_id)
+        if const_node is not None:
+            return True
+
+    return False
+
+
+def _collect_ops_with_weights(model):
+    ops_with_weights = []
+    for op in model.get_ops():
+        if op.get_type_name() == "MatMul":
+            constant_node_0 = _get_operation_const_op(op, const_port_id=0)
+            constant_node_1 = _get_operation_const_op(op, const_port_id=1)
+            if constant_node_0 or constant_node_1:
+                ops_with_weights.append(op.get_friendly_name())
+        if op.get_type_name() == "Gather" and _is_embedding(op):
+            ops_with_weights.append(op.get_friendly_name())
+
+    return ops_with_weights
+
+
+def get_stable_diffusion_dataset(
+    dataset_name: str, nsamples: int = 50, seed: int = 0, text_column: str = "caption"
+) -> nncf.Dataset:
+    if dataset_name not in ["conceptual_captions", "laion/220k-GPT4Vision-captions-from-LIVIS", "laion/filtered-wit"]:
+        raise ValueError(
+            f"""You have entered a string value for dataset. You can only choose between
+             ['conceptual_captions','laion/220k-GPT4Vision-captions-from-LIVIS','laion/filtered-wit'],
+             but we found {dataset_name}"""
+        )
+
+    data = load_dataset(dataset_name, split="train", streaming=True).shuffle(seed=seed).take(nsamples)
+    dataset = [batch[text_column] for batch in data]
+    return nncf.Dataset(dataset)
+
+
+def _hybrid_quantization(model: openvino.runtime.Model, quantization_config: Union[OVWeightQuantizationConfig, Dict]):
+    dataset = quantization_config.dataset
+    wc_ignored_scope = deepcopy(quantization_config.ignored_scope)
+
+    if isinstance(wc_ignored_scope, dict):
+        wc_ignored_scope["types"] = wc_ignored_scope.get("types", []) + ["Convolution"]
+    else:
+        assert wc_ignored_scope is None
```

would prefer to raise an error here (with explicit message)

---

I reformatted this part